lib/light-service/context.rb

```diff
@@ -8,16 +8,18 @@ module Outcomes
 
   # rubocop:disable ClassLength
   class Context < Hash
-    attr_accessor :message, :error_code, :current_action
+    attr_accessor :message, :error_code, :current_action, :orchestrator_callbacks
 
     def initialize(context = {},
                    outcome = Outcomes::SUCCESS,
                    message = '',
-                   error_code = nil)
+                   error_code = nil,
+                   orchestrator_callbacks = {})
```

Comment:
Storing the lambdas in `context.orchestrator_callbacks[:after_failing]` to leave the door open to other kinds of callbacks later.
Also, `orchestrator_callbacks` and not `callbacks` in order to disambiguate with the `with_callback` feature.